util/db/dbutil_test.go

```diff
@@ -255,28 +257,41 @@ func TestDBConcurrencyRW(t *testing.T) {
 
 	started := make(chan struct{})
 	var lastInsert int64
-	var wg sync.WaitGroup
-	wg.Add(1)
+	testRoutineComplete := make(chan struct{}, 3)
+	targetTestDurationTimer := time.After(15 * time.Second)
 	go func() {
-		defer wg.Done()
-		defer atomic.StoreInt64(&lastInsert, -1)
-		for i := int64(0); i < 10000; i++ {
-			errw := acc.Atomic(func(tx *sql.Tx) error {
+		defer func() {
+			atomic.StoreInt64(&lastInsert, -1)
+			testRoutineComplete <- struct{}{}
+		}()
+		var errw error
+		for i := int64(1); i < 10000 && errw == nil; i++ {
```

Comment:
Now that you have `targetTestDurationTimer`, maybe it's reasonable to get rid of the `10000` constant altogether?